src/main/webapp/app/grading-system/grading-system.component.html

```diff
@@ -57,21 +79,45 @@ <h2 style="margin-bottom: 0.5em">
                 </tr>
             </thead>
             <tbody>
-                <tr *ngFor="let gradeStep of gradingScale.gradeSteps; let i = index">
+                <tr *ngFor="let gradeStep of gradingScale.gradeSteps; let i = index" class="grading-scale-table-row">
+                    <td>
+                        <input [(ngModel)]="gradeStep.lowerBoundPercentage" type="number" step="0.01" min="0" max="100" required (change)="setPoints(gradeStep, true)" />
+                    </td>
+                    <td>
+                        <input
+                            [(ngModel)]="gradeStep.lowerBoundPoints"
+                            type="number"
+                            step="0.01"
+                            min="0"
+                            [max]="getMaxPoints() || 999"
+                            [required]="maxPointsValid()"
+                            [disabled]="!maxPointsValid()"
+                            (change)="setPercentage(gradeStep, true)"
+                        />
+                    </td>
                     <td>
-                        <input [(ngModel)]="gradeStep.lowerBoundPercentage" type="number" min="0" max="100" required />
+                        <input [(ngModel)]="gradeStep.upperBoundPercentage" type="number" step="0.01" min="0" max="100" required (change)="setPoints(gradeStep, false)" />
                     </td>
                     <td>
-                        <input [(ngModel)]="gradeStep.upperBoundPercentage" type="number" min="0" max="100" required />
+                        <input
+                            [(ngModel)]="gradeStep.upperBoundPoints"
+                            type="number"
+                            step="0.01"
+                            min="0"
+                            [max]="getMaxPoints() || 999"
+                            [required]="maxPointsValid()"
+                            [disabled]="!maxPointsValid()"
+                            (change)="setPercentage(gradeStep, false)"
+                        />
                     </td>
                     <td *ngIf="isGradeType()">
                         <input [(ngModel)]="gradeStep.gradeName" type="text" required />
                     </td>
                     <td *ngIf="!isGradeType()">
-                        <input [(ngModel)]="gradeStep.gradeName" type="number" step="0.01" min="0" placeholder="0" required />
+                        <input [(ngModel)]="gradeStep.gradeName" type="number" step="0.25" min="0" placeholder="0" required />
```

Comment:
I don't understand this. Why do we use number in case it is not a grade type? (the method name is not very intuitive, what is a grade type?)

Reply:
I removed the method, now it just checks if  `gradingScale.gradeType != GradeType.GRADE`. Because the bonus points are a number, and I wanted to use the built-in validation to enforce that.